airflow/providers/cncf/kubernetes/backcompat/volume_mount.py

```diff
@@ -0,0 +1,62 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Classes for interacting with Kubernetes API"""
+
+import warnings
+
+from kubernetes.client import models as k8s
+
+warnings.warn(
+    "This module is deprecated. Please use `kubernetes.client.models.V1Volume`.",
```

Comment:
This needs to live as `from airflow.kubernetes.volume_mount import VolumeMount` to be a back-compat layer.

Same for all the others.

Reply:
Not the deprecation message, but this file. Calling it airflow.providers.cncf.kubernetes.backcompat.volume_mount won't help users.

Reply:
@ashb  we can actually just change the import path using bowler for backport_providers. I was asked by @kaxil and @potiuk to set this up exclusively in the provider package. This will allow us to upgrade these as needed without releasing new airflow versions.

Reply:
See https://github.com/apache/airflow/pull/12384#discussion_r524534519